### PR TITLE
Place all the OpenShift related logic at one place

### PR DESCRIPTION
**Pull request by fridex**

### CODE REVIEW

File: thoth/common/openshift.py [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+        """It is expected that there is only one object type for the given item."""
+        if len(response.items) != 1:
+            raise RuntimeError(
+                f"Application misconfiguration - number of templates available in the infra namespace "
+                f"{self.infra_namespace!r} is {len(response.items)}, should be 1."
+            )
+
+    def _oc_process(self, namespace: str, template: dict) -> dict:
+        """Process the given template in OpenShift."""
+        # This does not work - see issue reported upstream:
```
Comment (goern) [inline thoth/common/openshift.py]: could you put some @todo here and lets see what happends?!

File: thoth/common/openshift.py [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+                if entry['name'] == parameter_name:
+                    entry['value'] = str(parameter_value)
+                    break
+            else:
+                template['parameters'].append({
+                    'name': parameter_name,
+                    'value': str(parameter_value)
+                })
+
+    def run_sync(self, force_analysis_results_sync: bool = False, force_solver_results_sync: bool = False) -> str:
```
Comment (goern) [inline thoth/common/openshift.py]: this whole section of preparing the pod/job template looks complicated, why is that?

Comment (fridex) [reply thoth/common/openshift.py]: It runs job and reuses all the configuration of cronjob.

**Comment by fridex** [comment]
@goern I would like to proceed with this one, but it looks like CI does not report status anymore. Could you take a look?

**Comment by fridex** [comment]
recheck

**Comment by fridex** [comment]
recheck

**Comment by fridex** [comment]
I will merge this to continue with work. If there are any issues/objections/suggestions I will incorporate them :)

**Comment by fridex** [comment]
recheck

**Pull request merged**